Packs/PAN-OS/Integrations/Panorama/Panorama_test.py

```diff
@@ -6542,9 +6532,81 @@ def test_second_fetch_with_two_incidents_with_different_log_types_and_different_
 
         assert incident_entries_dict == expected_parsed_incident_entries
         assert last_fetch_dict.get('X_log_type', '') == '2022-01-01 13:00:00'
-        assert last_id_dict.get('X_log_type', '') == '000000002'
+        assert last_id_dict.get('X_log_type', '') == {'dummy_device1': '000000002'}
         assert last_fetch_dict.get('Y_log_type', '') == '2022-01-01 13:00:00'
-        assert last_id_dict.get('Y_log_type', '') == '000000002'
+        assert last_id_dict.get('Y_log_type', '') == {'dummy_device2': '000000003'}
+
+
+def test_update_max_fetch_dict(mocker):
+    """
+    Given:
+    - configured_max_fetch - maximum number of incidents the client configured to fetch.
+    - max_fetch_dict - dictionary with the maximum number of incidents to fetch per log type.
+    - last_fetch_dict - dictionary with the last fetch time per log type.
+    When:
+        - update_max_fetch_dict is called.
+        - last_fetch_dict has a log type that is not in max_fetch_dict.
+    Then:
+        - The max_fetch_dict is updated with the maximum number of incidents to fetch per log type.
+    """
+    from Panorama import update_max_fetch_dict
+    mocker.patch('demistomock.getLastRun', return_value={"last_fetch_dict": {"log_type1": "2023-05-01 07:22:08"}})
+    res = update_max_fetch_dict(configured_max_fetch=5, max_fetch_dict={"log_type1": 10, "log_type2": 15},
+                                last_fetch_dict={"log_type1": "2023-05-01 07:22:08", "log_type2": "2023-05-01 07:22:08"})
+    assert res == {'log_type1': 15, 'log_type2': 5}
+
+
+def test_find_largest_id_per_device():
+    """
+    Given:
+    - list of dictionares repesenting raw entries.
+    When:
+        - find_largest_id_per_device is called.
+    Then:
+        - return a dictionary with the largest id per device.
+    """
+    raw_entries = [{'device_name': 'dummy_device1', 'seqno': '000000001'},
+                   {'device_name': 'dummy_device1', 'seqno': '000000002'},
+                   {'device_name': 'dummy_device2', 'seqno': '000000001'}]
+    from Panorama import find_largest_id_per_device
+    res = find_largest_id_per_device(raw_entries)
+    assert res == {'dummy_device1': '000000002', 'dummy_device2': '000000001'}
+
+
+def test_remove_duplicate_entries():
+    """
+    Given:
+    - list of dictionares repesenting raw entries, some contain seqno and some not.
+    - dictionary with the largest id per device.
+    When:
+    - remove_duplicate_entries is called.
+    Then:
+    - return a dictionary the entries that there id is larger then the id in the id_dict, and the entries that do not have seqno.
```

Comment:
why do we choose the entries that don't have a seqno? let's talk about it cause I'm not sure I understand.